python/python/lance/dataset.py

```diff
@@ -1598,6 +1609,35 @@ def optimize_indices(self, **kwargs):
         self._dataset._ds.optimize_indices(**kwargs)
 
 
+class DatasetStats(TypedDict):
+    num_deleted_rows: int
+    num_fragments: int
+    num_small_files: int
+
+
+class IndexStatS(TypedDict):
+    num_indexed_rows: int
+    num_unindexed_rows: int
```

Comment:
On second thought, given there are extra fields for the index stats, maybe we should leave this un-typed for now.